backend/src/apiserver/resource/resource_manager.go

```diff
@@ -286,6 +286,8 @@ func (r *ResourceManager) CreateRun(apiRun *api.Run) (*model.RunDetail, error) {
 
 	// Disable istio sidecar injection
 	workflow.SetAnnotationsToAllTemplates(util.AnnotationKeyIstioSidecarInject, util.AnnotationValueIstioSidecarInjectDisabled)
+	// Add a KFP specific annotation for cache service filtering
+	workflow.SetAnnotationsToAllTemplates(util.AnnotationKeyCacheEnabled, util.AnnotationValueCacheEnabled)
```

Comment:
It might be beneficial to use a label here instead of annotation. Labels can be used by Kubernetes-native filtering (for example you can request to watch only pods containing some label).